gobblin-core/src/main/java/gobblin/metrics/GobblinMetricsRegistry.java

```diff
@@ -54,7 +53,7 @@ public GobblinMetrics get(String id) {
    *         instance for the given job is not found
    */
   public GobblinMetrics remove(String id) {
-    return this.metricsMap.remove(id);
+    return this.metricsMap.asMap().remove(id);
```

Comment:
invalidate does not return the object upon removal, so we would have to change API of the method, or put additional logic here to first get the object to return.